### PR TITLE
Feat-Go: add Lcfirst, Strchr function

### PR DESCRIPTION
**Pull request by YogiPristiawan**
- Programming langugage: 
- Function(s) that I'm implementing:
- Issue number: 

- [x] I have created unit tests to support my implementation
- [x] I have created in-code documentation (or docstring) to document my code
- [ ] (Optional) I have updated the README.md file to document the function I'm implementing

### CODE REVIEW

File: go/lcfirst.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import "unicode"
+
+// The Lcfirst() function converts the first character of a string to lowercase.
+// Parameter
+// - str => string to be converted
+// Return
+// - res => converted string
+func Lcfirst(str string) string {
+	for _, val := range str {
```
Comment (elianiva) [inline go/lcfirst.go:11]: kenapa harus pake loop? ngga ambil index 0 aja?

Comment (YogiPristiawan) [reply go/lcfirst.go:11]: buat prevent kalau parameternya empty string. hmm mungkin masih bisa direfactor ya. kucoba dulu

File: go/strchr.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	if val, ok := search.(int); ok {
+		searchedCharacter = string(rune(val))
+	} else {
+		if val, ok := search.(string); ok {
+			searchedCharacter = val
+		} else {
+			return "", ErrStrchrInvalidParameterType
+		}
+	}
```
Comment (aldy505) [inline go/strchr.go:34]: jangan gini. gini aja:

```go
switch val := search.(type) {
case int:
  // handle
case string:
 // handle
default:
  // handle
}
```

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: go/lcfirst.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+// The Lcfirst() function converts the first character of a string to lowercase.
+// Parameter
+// - str => string to be converted
+// Return
+// - res => converted string
+func Lcfirst(str string) string {
+	if len(str) == 0 {
+		return ""
+	}
```
[the remark this answers is not in the document]
Comment (YogiPristiawan) [reply go/lcfirst.go:13]: oke noted

File: go/lcrifst_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+			},
+			{
+				param:  "golang",
+				expect: "golang",
+			},
+			{
+				param:  "G",
+				expect: "g",
+			},
+		}
```
[the remark this answers is not in the document]
Comment (YogiPristiawan) [reply go/lcrifst_test.go]: done

**Review by elianiva — APPROVED** [review]

**Pull request merged**